src/polynomial/lagrange.rs

```diff
@@ -0,0 +1,88 @@
+use std::{iter, num::NonZeroU32};
+
+use ff::PrimeField;
+
+use crate::fft;
+
+/// Lazy eval the values of the Lagrange polynomial for a cyclic subgroup of length `n` (`2.pow(log_n)`) at
+/// the `challenge` point
+///
+/// # Mathematical Representation
+///
+/// ```math
+/// L_i(X)=\frac{\omega^i}{n}\frac{X^n-1}{X-\omega^i}
+/// ```
+/// where {1, \omega, \omega^2, ..., \omega^n} - cyclic group
```

Comment:
add comment about what is the input and what are the output.
```
/// input: 
/// x: point to be evaluated
/// log_n: log of cyclic group size
/// output: {L_i(x)}, i in {0,...,n-1}
```

Reply:
Done